app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -4,14 +4,65 @@
 
 public class CalculatorModel implements CalculatorContract.CalculatorModelContract {
 
-    private String firstOperand;
+    private String firstOperand = "";
+    private String secondOperand = "";
+    private String operator = "";
+    private String result = "";
+
 
     @Override
     public void saveNumber(String number) {
-        firstOperand = number;
+        if (operator.isEmpty()) {
+            firstOperand = firstOperand + number;
+        } else {
+            secondOperand = secondOperand + number;
+        }
+    }
+
+    public String getPartialResult() {
+
+        return (firstOperand + operator + secondOperand);
+    }
+
+    @Override
+    public String getFullResult() {
+
+        switch (operator) {
+
+            case "+":
+                result = String.valueOf((Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand)));
+                break;
+
+            case "-":
+                result = String.valueOf((Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand)));
+                break;
+
+            case "/":
+                result = String.valueOf((Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand)));
+                break;
+
+            case "X":
+                result = String.valueOf((Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand)));
+                break;
+
+            default:
+                throw new IllegalStateException("Error");
```

Comment:
Instead of throw an exception, that you are no catching, and will cause a crash. please show a message on the screen or something more friendly.

Reply:
Done. thx